src/goto-symex/goto_symex_state.h

```diff
@@ -340,9 +340,13 @@ class goto_symex_statet final
   bool l2_thread_read_encoding(ssa_exprt &expr, const namespacet &ns);
   bool l2_thread_write_encoding(const ssa_exprt &expr, const namespacet &ns);
 
+  void populate_dirty_for_function(
+    const irep_idt &id, const goto_functionst::goto_functiont &);
+
   void switch_to_thread(unsigned t);
   bool record_events;
-  std::unique_ptr<const dirtyt> dirty;
+  std::unordered_set<irep_idt, irep_id_hash> dirty_processed_functions;
```

Comment:
To me, this is the wrong place: this belong's to the state of the symbolic executor (`goto_symext`) and is not part of the symbolic program state.